app/controllers/restaurants_controller.rb

```diff
@@ -1,9 +1,29 @@
 class RestaurantsController < ApplicationController
+  load_and_authorize_resource # potentially refactor this into ApplicationController
+
   def index
   end
 
   def new
     @restaurant = Restaurant.new
   end
 
+  def create
+    @restaurant = Restaurant.new(restaurant_params.merge({user: current_user}))
+    if @restaurant.save
+      render :show
+    else
+      flash[:alert] = @restaurant.errors.full_messages.first
+      render :new
+    end
+  end
```

Comment:
The creation of a restaurant should be limited if the user already has a restaurant